src/test/java/org/opensearch/integTest/PromptHandler.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.integTest;
+
+import com.google.gson.annotations.SerializedName;
+
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+public class PromptHandler {
+
+    boolean apply(String prompt) {
+        return prompt.contains(toolInput().getQuestion());
+    }
+
+    ToolInput toolInput() {
+        return new ToolInput();
+    }
+
+    String response(String prompt) {
+        if (prompt.contains("TOOL RESPONSE: ")) {
```

Comment:
Will this `TOOL RESPONSE: ` be used when writing actual ITs for a tool? If so, should we extract it to a common constant?

Reply:
That depends. For most tools that use flow agent to do integration test, they will not interact with LLM. This default implementation is used to test conversational agent flow with mock LLM server. Based on this assumption, we can leave it as is.